File: src/main/java/cc/moecraft/logger/coloring/AnsiColorMode.java
package cc.moecraft.logger.coloring;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 此类由 Hykilpikonna 在 2018/12/16 创建!
 * Created by Hykilpikonna on 2018/12/16!
 * Github: https://github.com/hykilpikonna
 * QQ: admin@moecraft.cc -OR- 871674895
 *
 * @author Hykilpikonna
 */
@AllArgsConstructor
@Getter
public enum AnsiColorMode
{
    COLOR_RGB("2;%s;%s;%s"),
    COLOR_8BIT("5;%s");
    
    private final String format;
}

File: src/main/kotlin/cc/moecraft/logger/coloring/AnsiColorMode.kt
package cc.moecraft.logger.coloring

import org.hydev.toAnsi256
import org.hydev.toAnsi8
import java.awt.Color

enum class AnsiColorMode(val format: (Color) -> String)
{
    COLOR_RGB({ it.toAnsi256() }),
    COLOR_8BIT({ it.toAnsi8() });

    /**
     * Format a color into this color format with r, g, and b params.
     */
    fun format(r: Int, g: Int, b: Int) = format(Color(r, g, b))
}
